File: CS/EntitiesLookupWithEditing/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Columns;

namespace EntitiesLookupWithEditing {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        NorthwindEntities entities = new NorthwindEntities();

        private void Form1_Load(object sender, EventArgs e) {
            gridView1.Columns.AddField("ProductName").Visible = true;
            GridColumn column = gridView1.Columns.AddField("Categories");
            RepositoryItemLookUpEdit editor = (RepositoryItemLookUpEdit)gridControl1.RepositoryItems.Add("LookUpEdit");
            editor.DataSource = entities.Categories;
            editor.DisplayMember = "CategoryName";
            editor.Columns.Add(new DevExpress.XtraEditors.Controls.LookUpColumnInfo("CategoryName"));
            column.ColumnEdit = editor;
            column.Visible = true;
            gridControl1.DataSource = entities.Products;
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            entities.SaveChanges();
        }
    }
}

File: VB/EntitiesLookupWithEditing/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Columns

Namespace EntitiesLookupWithEditing

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private entities As NorthwindEntities = New NorthwindEntities()

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridView1.Columns.AddField("ProductName").Visible = True
            Dim column As GridColumn = gridView1.Columns.AddField("Categories")
            Dim editor As RepositoryItemLookUpEdit = CType(gridControl1.RepositoryItems.Add("LookUpEdit"), RepositoryItemLookUpEdit)
            editor.DataSource = entities.Categories
            editor.DisplayMember = "CategoryName"
            editor.Columns.Add(New DevExpress.XtraEditors.Controls.LookUpColumnInfo("CategoryName"))
            column.ColumnEdit = editor
            column.Visible = True
            gridControl1.DataSource = entities.Products
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs)
            entities.SaveChanges()
        End Sub
    End Class
End Namespace
